src/binder/bind/bind_copy.cpp

```diff
@@ -213,16 +204,29 @@ expression_vector Binder::bindExpectedNodeFileColumns(
             if (skipPropertyInFile(*property)) {
                 continue;
             }
-            readerConfig.columnNames.push_back(property->getName());
-            readerConfig.columnTypes.push_back(property->getDataType()->copy());
-            columns.push_back(createVariable(property->getName(), *property->getDataType()));
+            expectedColumnNames.push_back(property->getName());
+            expectedColumnTypes.push_back(property->getDataType()->copy());
         }
     } break;
     default: {
         throw NotImplementedException{"Binder::bindCopyNodeColumns"};
     }
     }
-    return columns;
+    if (readerConfig.fileType == common::FileType::TURTLE) {
+        // Nothing to validate for turtle
+        return createColumnExpressions(readerConfig, expectedColumnNames, expectedColumnTypes);
+    }
+    // Detect columns from file.
+    std::vector<std::string> detectedColumnNames;
+    std::vector<std::unique_ptr<common::LogicalType>> detectedColumnTypes;
+    sniffFiles(readerConfig, detectedColumnNames, detectedColumnTypes);
+    // Validate.
+    validateNumColumns(expectedColumnTypes.size(), detectedColumnTypes.size());
+    if (readerConfig.fileType == common::FileType::PARQUET) {
+        // HACK(Ziyi): We should allow casting in Parquet reader.
+        validateColumnTypes(expectedColumnNames, expectedColumnTypes, detectedColumnTypes);
+    }
+    return createColumnExpressions(readerConfig, expectedColumnNames, expectedColumnTypes);
```

Comment:
```suggestion
    // Nothing to validate for turtle
    if (readerConfig.fileType != common::FileType::TURTLE) {
        // Detect columns from file.
        std::vector<std::string> detectedColumnNames;
        std::vector<std::unique_ptr<common::LogicalType>> detectedColumnTypes;
        sniffFiles(readerConfig, detectedColumnNames, detectedColumnTypes);
        // Validate.
        validateNumColumns(expectedColumnTypes.size(), detectedColumnTypes.size());
        if (readerConfig.fileType == common::FileType::PARQUET) {
            // HACK(Ziyi): We should allow casting in Parquet reader.
            validateColumnTypes(expectedColumnNames, expectedColumnTypes, detectedColumnTypes);
        }
    }
    return createColumnExpressions(readerConfig, expectedColumnNames, expectedColumnTypes);
```

Reply:
ditto. Plus instead of writing nested if statement
```
if (conditionA) {
   doA;
   if (conditionB) {
       doB; return;
   }
}
doC; return;
```
I prefer
```
if (!conditionA) {
   doC; return;
}
doA;
if (conditionB){
    doB; return;
}
```